docs/4.x/working-with-piwiks-ui.md

```diff
@@ -167,6 +167,8 @@ Piwik Core defines many classes that should be reused by new plugins and contrib
 
 ### Broadcast
 
+This class is considered deprecated and only exists for historical reasons as some components are still using it. You should usually instead use angularjs components like [$location](https://docs.angularjs.org/api/ng/service/$location).
```

Comment:
`$location` actually doesn't always work the way we want. For example, it does not handle array parameters right (IIRC). I don't think we can deprecate it.

Reply:
It's more like discouraged to be used and angularjs is preferred whenever any possible. I'll make it more clear 👍